src/dhcp/server.c

```diff
@@ -675,7 +675,7 @@ static gboolean listener_event(GIOChannel *channel, GIOCondition condition,
 
 	server_id_option = dhcp_get_option(&packet, DHCP_SERVER_ID);
 	if (server_id_option) {
-		uint32_t server_nid = get_be32(server_id_option);
+		uint32_t server_nid = get_unaligned((uint32_t *) server_id_option);
```

Comment:
Since now the SERVER_ID option is in wright byte order, this line no longer need to change the byte order of SERVER_ID option before compare it to dhcp_server->server_nip (both in big endian), so replace get_be32() with get_unaligned().